### PR TITLE
F/logger

### PR DESCRIPTION
**Pull request by Onurbon**

### CODE REVIEW

File: packages/core/src/events.ts [hunk 1]
```diff
@@ -35,7 +37,7 @@ export const completeEvent = (event: Event, store: Store<any, any>): void => {
 
   store.universe = nextUniverse;
   event.nextUniverse = nextUniverse;
-  submitPatches(store, store.universe, event.patches);
+  submitPatches(store, store.universe, event.patches, event);
```
Comment (coffee-cup) [inline packages/core/src/events.ts]: No need to pass in both `event.patches` and `event` here. Can just pass in `event`

File: packages/logger-plugin/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { ProdoPlugin } from "@prodo/core";
+
+export interface LoggerConfig {
+  logger?: boolean;
+}
+
+const init = (config: LoggerConfig) => {
+  if (config.logger) {
+    // tslint:disable-next-line
+    console.log("@prodo/logger is on");
```
Comment (coffee-cup) [inline packages/logger-plugin/src/index.tsx]: We probably want to disable this in `NODE_ENV === "production"` mode

File: packages/core/src/watch.ts [hunk 3]
```diff
@@ -123,9 +124,11 @@ export const submitPatches = (
 
   logger.info("[update cycle]", comps);
 
+  event.rerender = {};
   Object.keys(comps).forEach(compId => {
     const { setState, name, newValues, status } = comps[compId];
     if (!status.unmounted) {
+      event.rerender![comps[compId].name] = true;
```
Comment (coffee-cup) [inline packages/core/src/watch.ts]: Why is this needed?

File: packages/logger-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@prodo/logger",
+  "version": "0.0.1",
+  "main": "lib/index.js",
+  "license": "MIT",
+  "private": true,
```
Comment (coffee-cup) [inline packages/logger-plugin/package.json]: Not sure if you want the logger plugin published yet, but if you do this line should be removed.

File: packages/core/src/watch.ts (hunk 3 above)
Comment (Onurbon) [reply packages/core/src/watch.ts]: so that we can log this to the console, and show which components are re-rendered as a result of a state update

File: packages/logger-plugin/package.json (hunk 4 above)
Comment (Onurbon) [reply packages/logger-plugin/package.json]: good catch!

File: packages/logger-plugin/src/index.tsx (hunk 2 above)
Comment (Onurbon) [reply packages/logger-plugin/src/index.tsx]: Users will generally want to disable the entire plugin in production, but I would probably leave that to them for now, rather than hardcoding it in the plugin itself.

Comment (coffee-cup) [reply packages/logger-plugin/src/index.tsx]: Right, thats a good point
